### PR TITLE
Loads global .env in vite.config to fix vscode extension

### PR DESCRIPTION
**Pull request by Rotorsoft**
## Link to Issue
Closes: #9897

This is the configuration that worked for me locally and using the VSCode extension. I had to disable parallel execution for the extension to be able to run all tests at once. We should continue working on trying to parallelize all tests in the future.

## Description of Changes
- Calls dotenv.config() in vitest.confit.ts to fix vscode extension failing to load root .env
- Removes unused env var INIT_TEST_DB
- Passes import.meta to database bootstrapping utils to keep track of caller - just logging now, but can be useful
- Removes vitest global setup. This is not needed since all seeders will bootstrap
- Centralizes parallel files/concurrent behavior in vitest.config 

## Test Plan
- Unit tested the `FIXME()` call.
- CA (click around) tested on local and frack:
  - TODO page
  - 

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

**Review by timolegros — DISMISSED** [review]

File: vite.config.ts [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 
-import path from 'path';
+import * as dotenv from 'dotenv';
 import { defineConfig } from 'vite';
 import tsconfigPaths from 'vite-tsconfig-paths';
 
+dotenv.config();
+
+const pkg = process.env.npm_package_name;
+console.log('vitest:', pkg);
```
Comment (kurtisassad) [inline vite.config.ts:10]: we should remove the console log

**Review by kurtisassad — APPROVED** [review]

File: vite.config.ts [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
+
 export default defineConfig({
   plugins: [tsconfigPaths()],
   test: {
-    globalSetup: path.resolve(
-      __dirname,
-      './libs/model/src/tester/vitestDatabaseSetup.ts',
-    ),
+    sequence: { concurrent: false },
+    fileParallelism: false, // pkg !== '@hicommonwealth/model',
```
Comment (timolegros) [inline vite.config.ts:16]: If you add this here then you can remove `--fileParallelism=false` from all the commands

File: vite.config.ts (hunk 1 above)
Comment (Rotorsoft) [reply vite.config.ts:10]: I would keep it as a hint for later refactoring, let's discuss tomorrow

**Comment by Rotorsoft** [comment]
Yes @timolegros, I'm using this opportunity to remove unused variables and simplify bootstrapping and configuration. Next step is to work on test parallelization (future PR)

**Review by rbennettcw — APPROVED** [review]

File: libs/model/test/email/digest-email-lifecycle.spec.ts [hunk 3]
```diff
@@ -14,7 +14,6 @@ describe('Digest email lifecycle', () => {
   let communityThree: z.infer<typeof Community> | undefined;
 
   beforeAll(async () => {
-    await bootstrap_testing(true);
```
[the remark this answers is not in the document]
Comment (Rotorsoft) [reply libs/model/test/email/digest-email-lifecycle.spec.ts:17]: I think this is a bad practice at the suite level, and just another thing to remember. By definition all tests should create their own sandbox of data without concerning about other tests (before or after) in order to achieve true parallelization. This is a first step towards the goal, and it doesn't seem to break anything so better to remove it before it's too late.

[the remark this answers is not in the document]
Comment (Rotorsoft) [reply libs/model/test/email/digest-email-lifecycle.spec.ts:17]: One idea I discussed with Ryan yesterday was to create a dedicated database for each lifecycle test (using the lifecycle name formatted for pg), and start moving all db related unit tests to lifecycles.

[the remark this answers is not in the document]
Comment (timolegros) [reply libs/model/test/email/digest-email-lifecycle.spec.ts:17]: > By definition all tests should create their own sandbox of data without concerning about other tests (before or after) in order to achieve true parallelization.

I agree that is why I am saying removing truncation here is bad. If you remove truncation in the tests there should be automatic top-level truncation to replace it no?

> One idea I discussed with Ryan yesterday was to create a dedicated database for each lifecycle test (using the lifecycle name formatted for pg), and start moving all db related unit tests to lifecycles.

We could just use test containers as Kurtis is doing for Anvil. Seems simpler than creating separate DBs on the same server.

[the remark this answers is not in the document]
Comment (Rotorsoft) [reply libs/model/test/email/digest-email-lifecycle.spec.ts:17]: > > By definition all tests should create their own sandbox of data without concerning about other tests (before or after) in order to achieve true parallelization.
> 
> I agree that is why I am saying removing truncation here is bad. If you remove truncation in the tests there should be automatic top-level truncation to replace it no?
> 

Truncation will interfere with other parallel tests. Ideally each test's data should be isolated, with no need to truncate the entire db



> > One idea I discussed with Ryan yesterday was to create a dedicated database for each lifecycle test (using the lifecycle name formatted for pg), and start moving all db related unit tests to lifecycles.
> 
> We could just use test containers as Kurtis is doing for Anvil. Seems simpler than creating separate DBs on the same server.

That works too, I'm for whatever is easier at this point

[the remark this answers is not in the document]
Comment (timolegros) [reply libs/model/test/email/digest-email-lifecycle.spec.ts:17]: By removing truncation you are making the tests dependent on each other. If I create a thread in one test suite and then count threads in another test suite that count would be dependent on the other test suite.

I am for parallelizing test suites and "sandboxing" the test suites but I don't understand how removing truncation is pushing us towards that goal (especially if in test truncation is not replaced with top-level per suite truncation). The way I see it, it is doing the opposite.

[the remark this answers is not in the document]
Comment (Rotorsoft) [reply libs/model/test/email/digest-email-lifecycle.spec.ts:17]: That's the point, counting threads should not depend on threads created by another test. Truncation as we have it is a brute force approach. I don't think we can make progress on parallel testing if we can truncate the entire db randomly.

[the remark this answers is not in the document]
Comment (timolegros) [reply libs/model/test/email/digest-email-lifecycle.spec.ts:17]: This is the way I see it: you say you want parallelized sandboxed suites and then remove the only thing that is keeping the suites sandboxed.

We should talk about this in a meeting - I don't understand your point.

**Pull request merged**